kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/MobiComKitBroadcastReceiver.java

```diff
@@ -41,14 +42,19 @@ public MobiComKitBroadcastReceiver(FragmentActivity fragmentActivity) {
     public void onReceive(Context context, Intent intent) {
         String action = intent.getAction();
         Message message = null;
-        String messageJson = intent.getStringExtra(MobiComKitConstants.MESSAGE_JSON_INTENT);
-        if (!TextUtils.isEmpty(messageJson)) {
-            message = (Message) GsonUtils.getObjectFromJson(messageJson, Message.class);
-
-            if (message != null) {
-                if ((hideActionMessages && message.isActionMessage()) || (message.isActionMessage() && TextUtils.isEmpty(message.getMessage()))) {
-                    message.setHidden(true);
-                }
+        String keyString = intent.getStringExtra("keyString");
+        if(!TextUtils.isEmpty(keyString)) {
+            message = new MessageDatabaseService(context).getMessage(keyString);
+        }
+        if(message == null) {
+            String messageJson = intent.getStringExtra(MobiComKitConstants.MESSAGE_JSON_INTENT);
```

Comment:
Do we need this getString? Are we setill passing Message JSON from other places?

Reply:
Ya we are passing it from other places too so it I think it is necessary otherwise after deleting message for everyone from the dashboard when the activity syncs the app crashes at `MobiComKitBroadcastReceiver line:87 conversationUIService.syncMessages(message, keyString);` .